bin/oneoff/back_populate_user_levels_level_source_id

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env ruby
+#
+# This script populates the `level_source_id` column of the `user_levels` table,
+# if not already present, using the most recent activities `level_source_id`.
+#
+# This is being done as a one-off script so as to be able to control the timing
+# of its execution in production. A migration will follow after the column is
+# populated in production so as to assure the column is populated in all
+# environments.
+
+require_relative '../../dashboard/config/environment'
+
+if ARGV.length != 4
+  puts 'Usage: ./bin/oneoff/back_populate_user_levels_level_source_id start_index end_index batch_size delay_sec'
+  exit 1
+end
+
+start_index = ARGV[0].to_i
+end_index = ARGV[1].to_i
+batch_size = ARGV[2].to_i
+delay_sec = ARGV[3].to_f
+
+until start_index > end_index
+  UserLevel.
+    where(id: start_index..(start_index + batch_size)).
+    where(level_source_id: nil).each do |user_level|
+
+    activity = Activity.
+      where(user_id: user_level.user_id, level_id: user_level.level_id).last
```

Comment:
For better or for worse, the `activities` table does not have `script_id` information.
```
DESCRIBE activities;
+-----------------+--------------+------+-----+---------+----------------+
| Field           | Type         | Null | Key | Default | Extra          |
+-----------------+--------------+------+-----+---------+----------------+
| id              | int(11)      | NO   | PRI | NULL    | auto_increment |
| user_id         | int(11)      | YES  | MUL | NULL    |                |
| level_id        | int(11)      | YES  |     | NULL    |                |
| action          | varchar(255) | YES  |     | NULL    |                |
| url             | varchar(255) | YES  |     | NULL    |                |
| created_at      | datetime     | YES  |     | NULL    |                |
| updated_at      | datetime     | YES  |     | NULL    |                |
| attempt         | int(11)      | YES  |     | NULL    |                |
| time            | int(11)      | YES  |     | NULL    |                |
| test_result     | int(11)      | YES  |     | NULL    |                |
| level_source_id | int(11)      | YES  | MUL | NULL    |                |
| lines           | int(11)      | NO   |     | 0       |                |
+-----------------+--------------+------+-----+---------+----------------+
```